config.go

```diff
@@ -106,16 +114,24 @@ func (c *Config) Validate() error {
 	if c.Group.Session.Timeout%time.Millisecond != 0 {
 		sarama.Logger.Println("Group.Session.Timeout only supports millisecond precision; nanoseconds will be truncated.")
 	}
-	if c.Group.PartitionStrategy != StrategyRange && c.Group.PartitionStrategy != StrategyRoundRobin {
-		sarama.Logger.Println("Group.PartitionStrategy is not supported; range will be assumed.")
-	}
 	if !c.Version.IsAtLeast(minVersion) {
 		sarama.Logger.Println("Version is not supported; 0.9. will be assumed.")
 		c.Version = minVersion
 	}
 	if err := c.Config.Validate(); err != nil {
 		return err
 	}
+	if c.Group.Balancer == nil {
+		switch c.Group.PartitionStrategy {
```

Comment:
if you want you could simplify this by first before the switch saying

```
c.Group.Balancer = Range
switch c.Group.PartitionStrategy {
  case StrategyRoundRobin:
     c.Group.Balancer = RoundRobin
  case StrategyRange: // noop
  default:
     	sarama.Logger.Println("Group.PartitionStrategy is not supported; range will be assumed.")
```

Reply:
Option 3: 
```
switch c.Group.PartitionStrategy {
case StrategyRoundRobin:
	c.Group.Balancer = RoundRobin
default:
	sarama.Logger.Println("Group.PartitionStrategy is not supported; range will be assumed.")
	fallthrough
case StrategyRange:
	c.Group.Balancer = Range
}
```
I think of the three of them, I prefer option 1, actually.  It feels more readable (to use a really subjective term) since each case is explicitly listed.  Though your suggestion is the least lines of code of the bunch. 😉